src/test/scala/com/github/simplesteph/ksm/parser/YamlAclParserTest.scala

```diff
@@ -0,0 +1,170 @@
+        |      'tr1*':
+        |        - All
+        |    clusters:
+        |      '*':
+        |        - Create,Allow,*
+        |  tom:
+        |    topics:
+        |      '*':
+        |        - Admin
+        |""".stripMargin
+
+    val res = yamlAclParser.aclsFromReader(new StringReader(yaml))
+
+    res.acls.map(_.toString()) shouldBe Set(
+      "(TransactionalId:PREFIXED:tr1,User:peter has Allow permission for operations: All from hosts: *)",
+      "(Cluster:LITERAL:*,User:peter has Allow permission for operations: Create from hosts: *)",
+      "(Cluster:LITERAL:*,User:alice has Allow permission for operations: Create from hosts: *)",
+      "(Topic:LITERAL:topic1,User:alice has Allow permission for operations: Read from hosts: *)",
+      "(Topic:LITERAL:topic1,User:alice has Allow permission for operations: Write from hosts: *)",
+      "(Topic:LITERAL:topic1,User:alice has Allow permission for operations: Describe from hosts: *)",
+      "(Topic:LITERAL:topic1,User:alice has Allow permission for operations: Create from hosts: *)",
+      "(Group:PREFIXED:mygroup-,User:alice has Deny permission for operations: Write from hosts: 12.34.56.78)",
+      "(Topic:LITERAL:*,User:tom has Allow permission for operations: Describe from hosts: *)",
+      "(Cluster:LITERAL:*,User:tom has Allow permission for operations: Create from hosts: *)",
+      "(Topic:LITERAL:*,User:tom has Allow permission for operations: Write from hosts: *)",
+      "(Topic:LITERAL:*,User:tom has Allow permission for operations: Delete from hosts: *)",
+      "(Topic:LITERAL:*,User:tom has Allow permission for operations: Read from hosts: *)",
+      "(Topic:LITERAL:*,User:tom has Allow permission for operations: Create from hosts: *)",
+      "(Group:LITERAL:group1,User:bob has Allow permission for operations: Write from hosts: *)",
+      "(Group:LITERAL:group1,User:bob has Allow permission for operations: Read from hosts: *)",
+      "(Group:LITERAL:group2,User:bob has Allow permission for operations: Read from hosts: *)",
+      "(Group:LITERAL:group2,User:bob has Allow permission for operations: Describe from hosts: *)")
+  }
+
+  "aclsFromYaml" should "catch all YAML errors" in {
+
+    val yaml =
+      """users:
+        |  alice:
+        |    groups:
+        |      mygroup-*:
+        |        - BadOperation,Deny,12.34.56.78
+        |    topics:
+        |      topic1: [ Consume, Produce ]
+        |  bob:
+        |    groups:
+        |      group1: [ BadOperation2, Write ]
+        |      group2:
+        |        - Describe,Allow,*
+        |        - Read
+        |  peter:
+        |    cluster:
+        |      '*':
+        |        - Create,Allow,*
+        |  tom:
+        |    topics:
+        |      '*':
+        |       - Admin
+        |""".stripMargin
+
+    val res = yamlAclParser.aclsFromReader(new StringReader(yaml))
+
+    res.acls.map(_.toString()).foreach(str => println(str))
+
+    res.errs.map(_.toString()) shouldBe List(
```

Comment:
again if we can test the data here inside of the `res.errs` and not the `.toString` that'd be great